Assets/HoloToolkit/Input/Scripts/Utilities/Managers/InputManager.cs

```diff
@@ -422,14 +425,16 @@ public void RaiseSourceUp(IInputSource source, uint sourceId, InteractionSourceP
             HandleEvent(inputEventData, OnSourceUpEventHandler);
 
             // UI events
-            if (ShouldSendUnityUiEvents && pressType == InteractionSourcePressInfo.Select)
+            IPointingSource pointingSource;
+            FocusManager.Instance.TryGetPointingSource(inputEventData, out pointingSource);
+            PointerInputEventData pointerInputEventData = FocusManager.Instance.GetSpecificPointerEventData(pointingSource);
+            if (pointerInputEventData != null && pressType == InteractionSourcePressInfo.Select)
             {
-                PointerInputEventData pointerInputEventData = FocusManager.Instance.GetPointerEventData();
                 pointerInputEventData.InputSource = source;
                 pointerInputEventData.SourceId = sourceId;
 
-                HandleEvent(pointerInputEventData, ExecuteEvents.pointerUpHandler);
-                HandleEvent(pointerInputEventData, ExecuteEvents.pointerClickHandler);
+                ExecuteEvents.ExecuteHierarchy(inputEventData.selectedObject, pointerInputEventData, ExecuteEvents.pointerUpHandler);
```

Comment:
When does inputEventData.selectedObject get set to something?

Reply:
During each pointer update

Reply:
Actually my last comment was wrong.  It's set on line 246 in the `InputManager`

```
            // Use focused object when OverrideFocusedObject is null.
            GameObject focusedObject = (OverrideFocusedObject == null) ? FocusManager.Instance.TryGetFocusedObject(eventData) : OverrideFocusedObject;
```

Reply:
That sets `focusedObject`, yeah, but where does `inputEventData.selectedObject` get updated?

Reply:
In the `TryGetFocusedObject(eventData)` method.

FocusManager line 350: `pointerInputEventData.selectedObject = details.Value.Object;`

Reply:
Thanks!